Dev

ITevgeniydav2022 opened the pull request:
Сдача ДЗ за 6 марта 2023г.

TatyanaBogodvid committed:
Созданы классы BotConfig,
BotInitializer,
TelegramBot.
Создана кнопка меню.

TatyanaBogodvid committed:
Merge branch 'dev' into tatiana-b

# Conflicts:
#	pom.xml
#	src/main/java/ru/team2/lookingforhouse/config/BotConfig.java
#	src/main/java/ru/team2/lookingforhouse/config/BotInitializer.java
#	src/main/resources/application.properties

ildar1902 committed:
внёс незначительные изменения (добавил две пустые строчки) чтоб сделать комит

ildar1902 committed:
внёс незначительные изменения (добавил две пустые строчки) чтоб сделать комит 2

TatyanaBogodvid committed:
Создано меню

TatyanaBogodvid committed:
Merge remote-tracking branch 'origin/dev' into dev

TatyanaBogodvid committed:
Добавлен коментарий к созданию меню.
Добавлен logback.xml.

TatyanaBogodvid committed:
Merge branch 'tatiana-b' into dev

gerarr9 committed:
Добавил базу данных и регистрацию пользователя

gerarr9 committed:
Добавил базу данных и регистрацию пользователя

gerarr9 committed:
Добавил базу данных и регистрацию пользователя

ildar1902 committed:
1. заменил версию зависимости telegrambots на 5.6.0
2. написал метод sendMessageToVolunteer, который отвечает за логику команды "вызвать волонтёра".
3. прописал команде /call_volunteer логику
4. внёс поправки в коде, например long chatId заменил на String chatId, немного оптимизировал метод sendMessage

ildar1902 committed:
Доработал метод "вызов волонтёра", добавил туда возможность вызова рандомного волонтёра.
Сделал метод генерации рандомного чат-айди.

TatyanaBogodvid committed:
Добавлены кнопки к сообщению /start, /info.
Изменен тип chatId long вместо String.
Изменено приветствие.
Добавлена общая информация о приюте.

TKaterinaS committed:
создала класс Animal, Cat, Dog и Java doc к ним.

TatyanaBogodvid committed:
Методы для клшек и собак написаны отдельно.
Нужно избавиться от кнопки меню, или переписать ее.

TatyanaBogodvid committed:
Merge branch 'dev' into tatiana-b

# Conflicts:
#	src/main/java/ru/team2/lookingforhouse/service/TelegramBot.java

TKaterinaS committed:
Изменила Java Doc,добавила поле chatId в классе User

TKaterinaS committed:
Merge remote-tracking branch 'origin/dev' into dev

KatOli4ka committed:
1.Создала репозитории для классов: юзер, животное, сошка, собака.
2. Создала контроллеры для классов: юзер, животное, сошка, собака.
3. Создала сервисы для классов: юзер, животное, сошка, собака и сервис для валидации данных по классам
4. Создала исключения(возможно и не понадобятся, поэтому они пока все закоментированны).
5. Создала класс для связи с фабрикой гибернета
6. Добавила зависимости в файл Pom.xml для работы с вэб-приложениями, сваггером, api

KatOli4ka committed:
1.Удалила не нужные папки валидации и гиберента и содежащиеся в них классы.
2. Добавила в application properties postgres dialect и sql show
3. Полностью передела логику методов контроллеров и сервисов через jpa репозиторий
4. Во всех классах, в которых я автор, отредактировала javadoc

KatOli4ka committed:
подредачила чуть-чуть методы сервиса update для всех классов

gerarr9 committed:
c-d

TatyanaBogodvid committed:
Добавлены кнопки как взять питомца из приюта и информация по ним.

TatyanaBogodvid committed:
Merge branch 'tatiana-b' into dev

ITevgeniydav2022 committed:
Откорректировал текст кнопок.

TatyanaBogodvid committed:
Изменен состав меню

TatyanaBogodvid committed:
Merge branch 'dev' into tatiana-b

# Conflicts:
#	src/main/java/ru/team2/lookingforhouse/service/TelegramBot.java

TatyanaBogodvid committed:
Исправлены некоторые опечатки

gerarr9 committed:
Adopter

gerarr9 committed:
Adopter

gerarr9 committed:
Merge remote-tracking branch 'origin/dev' into dev

gerarr9 committed:
UserCat and UserDog

gerarr9 committed:
Dodelal

gerarr9 committed:
Dodelal

KatOli4ka committed:
удалила не нужные классы, сервисы, контроллелы и репозитории

KatOli4ka committed:
Простите, что была не внимательна и написала код сразу в дэв.
1.Добавила исключения и репозитории к новым классам, а также javadoc

gerarr9 committed:
Dodelal

KatOli4ka committed:
Добавила поле чат-айди в классы ContactUserCat и ContactUserDog, репозитории к новым классам, а также javadoc

gerarr9 committed:
Dodelal

gerarr9 committed:
Dodelal

KatOli4ka committed:
Merge remote-tracking branch 'origin/dev' into dev

ITevgeniydav2022 committed:
Изменил название поля foto на photo в классе ReportData.

KatOli4ka committed:
Добавила аннотации в новые классы, создала сервисы к новым классам

KatOli4ka committed:
Переименование классов

KatOli4ka committed:
1.Вернула данные логгера для ноутов(не маков).
2.Добавила в класс report поле с id (необходимо его добавить в саму таблицу)
3.Отредактировала репозиторий report и создала сервиск к нему. Пока сырая версия

KatOli4ka committed:
1.Сделала метод sendMessage  публичным, чтобы использовать его в контроллере report
2. Создала контроллеры к новым классам. Контроллер report оч сырой - будет в дальнейшем дорабатываться

ITevgeniydav2022 committed:
Update README.md

ITevgeniydav2022 committed:
Update README.md

ildar1902 committed:
в классах, где есть поле registeredAt исправил опечатку и изменил тип данных вместо String на Timestamp.

в классе Report у поля photoId сменил тип данных с byte[] на String, потому что в таблице будет храниться не файл, а только его айдишник, а он типа String.

в классах UserCat и UserDog убрал поле id, основным полем с аннотацией @Id сделал поле chatId.

в классах сервисах внёс изменения там, где были пересечения с полями, которые отредактировал в предыдущих классах (у photoId сменил тип данных, id заменил на chatId).

самые большие изменения в классе TelegramBot.
Во-первых, создал отдельный класс Constant и кучу константных полей из класса TelegramBot вынес туда.
Во-вторых, убрал метод регистрации пользователя при выполнении команды старт, создал методы регистрации пользователей приюта собак и пользователей приюта кошек в соответствующие таблицы. Привязал эти методы соответствующим кнопкам CAT_BUTTON и DOG_BUTTON.
В-третьих, немного подредактировал код визуально в некоторых местах и поправил формат (Ctrl+Alt+L).

ildar1902 committed:
Merge branch 'ildar-a' into dev

ITevgeniydav2022 committed:
Update README.md

ITevgeniydav2022 committed:
Update README.md

ildar1902 committed:
добавил JavaDoc комментарии к методам регистрации пользователей приютов кошек/собак.
отредактировал JavaDoc комментарий к методу вызова волонтёра.
и пара незначительных изменений типа удаления неиспользуемых импортов и лишних строчек

ITevgeniydav2022 committed:
Update README.md

ildar1902 committed:
Временно закомментил кусок кода в классе ReportController, где выскакивало исключение при запуске.

Отредактировал реализацию команды "Позвать волонтёра". Сообщение пользователю о том, что с ним свяжутся, добавил в сам метод вызова волонтёра, тем самым, сообщение пользователю отправляется чуть быстрее после нажатия кнопки "Позвать волонтёра".

TatyanaBogodvid committed:
Merge branch 'dev'

ildar1902 committed:
Внёс небольшие косметические изменения) If-else-if заменил на Switch, поправил комментарии

ildar1902 committed:
Удалил 25 классов. Создал Enum UserStatus. Отредактировал сущности UserCat, UserDog и Report.
Отредактировал метод регистрации пользователя, добавил туда присвоение статуса новому пользователю (по умолчанию это "простой пользователь").

KatOli4ka committed:
Навела косметику в существующих репозиториях, сервисах и контроллерах

TatyanaBogodvid committed:
Merge remote-tracking branch 'origin/dev' into dev

TatyanaBogodvid committed:
Добавлен шаблон отчета

ildar1902 commented on src/main/java/ru/team2/lookingforhouse/model/UserDog.java:
```diff
@@ -0,0 +1,24 @@
+    private String phoneNumber;
+    private String email;
+    private Timestamp registeredAt;
+    private String userStatus;
```
так у нас создан enum. в методе, где мы сохраняем данные в таблицу, мы используем enum значение (посмотреть это можно в классе TelegramBot, строка 846), просто приписываем к нему toString(),  чтобы в таблице хранилась строка.  ты говорил, что в таблице enum нужно хранить в виде строки. как правильнее это сделать?

ildar1902 replied:
или поле лучше оставить типа UserStatus, а в методе прописывать String.valueof()? 🤔

ildar1902 replied:
вернее, получается, наоборот. UserStatus.valueOf(JUST_USER.toString()).

ildar1902 replied:
спасибо

ildar1902 replied:
а можно как-то сделать, чтобы в таблице было не само название значения, а его строчный вариант? у нас например значение JUST_USER имеет строчное значение "простой пользователь". это как то в аннотации можно поколдовать?)

ildar1902 replied:
🤝спасибо👍

ildar1902 committed:
поправил поле userStatus в классах UserDog и UserCat, как Дима прокомментировал, поменял тип данных String на enum UserStatus, над полем поставил соответствующую аннотацию, чтобы значение а твблице хранилось в строке.

ildar1902 committed:
Создал конвертер для поля UserStatus. Заменил аннотацию @Enumerated на аннотацию @Convert. Теперь в классе UserCat и UserDog поле userStatus имеет тип enum UserStatus, а в таблицу сохраняется строка в нужном нам виде

ildar1902 committed:
В классе TelegramBot и в классе BotInitializer сделал инъекции, как положено, через конструктор.

TatyanaBogodvid committed:
Merge remote-tracking branch 'origin/dev' into dev

TatyanaBogodvid committed:
Merge branch 'tatiana-b' into dev

ildar1902 committed:
Заменил Report на ReportCat и ReportDog, чтобы нормально установить связи с сущностями UserCat и UserDog. В связи с этим, сущность Report и все связанные с ней классы пришлось удалить.

ildar1902 committed:
Merge branch 'ildar-a' into dev

ildar1902 committed:
В сущностях UserCat и UserDog убрал поле email, будем сохранять только номер телефона для связи

ildar1902 committed:
Merge branch 'ildar-a' into dev

ildar1902 committed:
Поправил ошибку в ReportDog. тип данных поля userDog сменил с UserCat на UserDog

KatOli4ka committed:
1.Создала контроллеры,сервисы, отредачила репозиторий для репортов.
2.Контроллеры для собаки не работают совсем, хотя у кота такие же контроллеры работают норм. Я не знаю как это исправить

KatOli4ka committed:
Исправила ошибку в контроллерах репорта. Все запускается

KatOli4ka committed:
Отредачила репозитории, сервисы и контроллеры всех классов. Добавила новые методы в них, удалила не нужные

KatOli4ka committed:
Подредачила методы, где используется сэт и коллекция во всех классах

ildar1902 committed:
Написан первый успешный тест на обработку команды старт

ildar1902 committed:
Написано ещё 3 теста - итого 4. Проверки обработки команд /start, /dog, /cat, /call_volunteer. В классе TelegramBot метод generateRandomChatId() сделал public, чтобы использовать его в тестах.

ildar1902 committed:
Продолжаю писать тесты, 11 штук готово. В класс Constant добавил несколько констант. Укоротил ссылки схемы проезда к приютам.

ITevgeniydav2022 committed:
Обработал кнопку "Сохранить контакт".

ITevgeniydav2022 committed:
Merge remote-tracking branch 'origin/dev' into dev

KatOli4ka committed:
Подредачила названия некоторых методов. Поменяла логер под обычных смертных

KatOli4ka committed:
подправила пунктуацию и орфографию в новых методах

ildar1902 committed:
Класс TelegramBot практически полностью покрыт тестами. Осталась только часть с методом добавления номера телефона. По нему пока не стал писать тест, потому что сам метод нужно немного доработать.

ildar1902 committed:
Установил модификатор доступа private полю BotConfig в классе TelegramBot. Уже не первый раз меняю, кто-то обнуляет это изменение постоянно из-за некорректной работы с гитом

KatOli4ka committed:
1.добавила аннотации во все модели
2.исправила javadoc для исключений

KatOli4ka committed:
Merge branch 'kate-o' into dev

KatOli4ka committed:
убрала лишнюю аннотацию

KatOli4ka committed:
Merge branch 'kate-o' into dev

ildar1902 committed:
подкорректировал сущности, убрал лишние аннотации

ildar1902 committed:
пофиксил кнопку сохранения контактов

ildar1902 committed:
сделал функцию сохранения отчётов. дописал тесты для класса TelegramBot

ildar1902 committed:
дописал тесты для всех классов пакета сервис.

ildar1902 committed:
добавил тесты, увеличил процент покрытия
